config/provider/upgrade/config.yml

```diff
@@ -0,0 +1,10 @@
+providers:
+- providername: provider-gcp
+  initialproviderpackage: crossplane/provider-gcp:v0.16.0
```

Comment:
How would you feel about just making these fields the tags, then making `providername` into `crossplane/provider-gcp`?

I also wonder if we should move towards a single config file for all provider "conformance" testing. This could look something like moving to `config/provider/conformance.yaml` and having a format like:
```
providers:
- package: crossplane/provider-gcp
  upgrade:
    - initial: v0.16.0
      final: master
    - initial: v0.15.0
      final: v0.16.0
...
```
This would allow us to continue to expand the attributes of providers that we deem necessary for conformance :)

Reply:
That seems like a good idea. This format looks good to me as well. Just wondering, will it be a good idea to keep the name of the provider as a separate field or should we extract the name from the package field by using `strings.SplitAfter(s, "/")`

Reply:
We can technically name the `Provider` anything we want, but using the last text group after `/` sounds like a good default. Since the name isn't really important to the actual test being performed I think it makes sense to eliminate it from configuration values :)

In regards to the upgradeable versions, I think we will likely want a variety of upgrade tests per provider as they determine their own release schedules. For instance, in `provider-gcp` supported the last 3 releases in the future, we may want to continue to test all the patches for each supported version (e.g. `v0.22.5` --> `v0.23.3`, `v0.23.3` --> `v0.24.1`, etc.)